### ISSUE TITLE
Output shape of the critic?

### ISSUE THREAD

**Issue by huiminzeng** [abacusai/intraprocessing_debiasing#3]
Hello,

I think the output the critic should be a vector of size batch_size * 1, and the cross entropy loss is a scalar. Is it intened that they are directly multiplied together? I couldn't perform backpropgation

https://github.com/abacusai/intraprocessing_debiasing/blob/b4f0c35e299022b1e71e26686220e90440687100/intraproc_celeba.py#L641

**Comment by yashsavani** [comment]
Hi,

Thanks for reaching out! In the adversarial model, we consider the critic to be a function of the entire batch. Unfortunately, the bias of a model cannot be assessed on a single example, so the loss of the critic outputs a single scalar for the entire batch. This scalar can then be multiplied with the cross-entropy loss to give a single combined scalar loss.

I hope that makes sense. If there is anything that is unclear please let me know.

**Comment by huiminzeng** [comment]
oh! i see your point now. Thank you very much!
